applications/tari_testnet_miner/src/miner.rs

```diff
@@ -111,12 +114,25 @@ impl Miner {
         }
         let interval = self.block.as_ref().unwrap().header.timestamp.timestamp() - old_header.timestamp.timestamp();
         let difficulty = Difficulty::calculate_req_difficulty(interval, self.difficulty);
-        let nonce = BlakePow::mine(difficulty, &self.block.as_ref().unwrap().header);
+        let nonce = BlakePow::mine(difficulty, &self.block.as_ref().unwrap().header, self.stop_mine.clone());
         self.block.as_mut().unwrap().header.nonce = nonce;
         self.difficulty = difficulty;
         Ok(())
     }
 
+    /// This gets the arc pointer to the mining control flag
+    pub fn get_mine_flag_ref(&self) -> Arc<AtomicBool> {
```

Comment:
Doesn't return a ref
```suggestion
    pub fn get_mine_flag(&self) -> Arc<AtomicBool> {
```

Reply:
An arc does return a pointer in a sense. But I am not too stuck on the name.